### DISCUSSION TITLE
Sending emails from a task through a script

### DISCUSSION THREAD

**Discussion by RolandvonBorstel** [sadieblue/agilityblue.workspace-scripts#10] [Q&A] [answered]
Do you have a code sample on how to use the Add-TaskEmailMessage command to generate an email in a task that can be sent via the Send-EmailMessage command?

**Comment by bmward80** [comment, marked as answer]
For sending emails, you can follow these steps:

1) First create a new email using the `Add-EmailMessage` cmdlet (separate multiple recipients with a semicolon). You can also add `Cc` and `Bcc` recipients if desired.

```powershell
$emailMessage = @{
    To = "someone1@somewhere.com; someone2@somewhere.com"
    FromAddress = "someone@somewhere.com"
    FromName = "Someone's Name"
    Subject = "Email created froma a script"
    Body = "<p>This is an email created from a script.</p><p>It accepts html that email clients allow.</p>"
}

$createdEmailMessage = Add-EmailMessage $emailMessage
```

2) Add contact relationships to the email with `Add-EmailMessageContacts` cmdlet (this is important for the UI to populate the To/Cc/Bcc fields when replying to the email)

```powershell
# Relationship Types:
# ----------------------------------------------
# 1 = To
# 3 = CC
# 4 = BCC
# ----------------------------------------------

# The list of recipients (contacts must already exist prior to doing this step and you need to know their ids)
$emailMessageContacts = @(
  @{
    EmailMessageId = $createdEmailMessage.EmailMessageId
    ContactId = 1 
    RelationshipTypeId = 1
  }
)

Add-EmailMessageContacts $emailMessageContacts | Out-Null
```

3) Create a task email with the `Add-TaskEmailMessage` cmdlet to link the email with an existing task

```powershell
$taskEmailMessage = @{
  TaskId = 29877
  EmailMessageId = $createdEmailMessage.EmailMessageId
}

# Create the link
Add-TaskEmailMessage -Entry $taskEmailMessage | Out-Null
```

4) Send the email using the `Send-EmailMessage` cmdlet

```powershell
Send-EmailMessage -Id $createdEmailMessage.EmailMessageId | Out-Null
```

Brandon

**Comment by RolandvonBorstel** [comment]
Hi Brandon. Thanks for the sample. That really helps a lot.
